### PR TITLE
Fix issue 9

### PR DESCRIPTION
**Pull request by londumas**
fix issue 9, store the number of pairs in the cross-correlation

### CODE REVIEW

File: py/pylya/xcf.py [hunk 1]
```diff
@@ -93,8 +98,9 @@ def fast_xcf(z1,r1,w1,d1,z2,r2,w2,ang):
     crp = sp.bincount(bins,weights=rp*we)
     crt = sp.bincount(bins,weights=rt*we)
     cz = sp.bincount(bins,weights=z*we)
+    cnb = sp.bincount(bins,weights=sp.ones(z.size))
```
Comment (ngbusca) [inline py/pylya/xcf.py]: remove keyword weights, they are 1 by default

File: py/pylya/xcf.py [hunk 2]
```diff
@@ -54,20 +58,21 @@ def xcf(pix):
             zqso = [q.zqso for q in d.neighs]
             we_qso = [q.we for q in d.neighs]
 
-            cw,cd,crp,crt,cz = fast_xcf(d.z,d.r_comov,d.we,d.de,zqso,rc_qso,we_qso,ang)
+            cw,cd,crp,crt,cz,cnb = fast_xcf(d.z,d.r_comov,d.we,d.de,zqso,rc_qso,we_qso,ang)
             
             xi[:len(cd)]+=cd
             we[:len(cw)]+=cw
             rp[:len(crp)]+=crp
             rt[:len(crp)]+=crt
```
Comment (ngbusca) [inline py/pylya/xcf.py]: this should be rt[:len(crt)] += crt

File: py/pylya/xcf.py [hunk 3]
```diff
@@ -54,20 +58,21 @@ def xcf(pix):
...
             z[:len(crp)]+=cz
```
Comment (ngbusca) [inline py/pylya/xcf.py]: this should be z[:len(cz)]+=cz

File: py/pylya/xcf.py [hunk 4]
```diff
@@ -54,20 +58,21 @@ def xcf(pix):
...
+            nb[:len(crp)]+=cnb
```
Comment (ngbusca) [inline py/pylya/xcf.py]: this should be nb[:len(nb)]+=cnb

File: bin/do_xcf.py [hunk 5]
```diff
@@ -182,11 +182,13 @@ def corr_func(p):
     rps=cfs[:,2,:]
     rts=cfs[:,3,:]
     zs=cfs[:,4,:]
+    nbs=cfs[:,5,:]
     cfs=cfs[:,1,:]
 
     rp = (rps*wes).sum(axis=0)/wes.sum(axis=0)
     rt = (rts*wes).sum(axis=0)/wes.sum(axis=0)
     z = (zs*wes).sum(axis=0)/wes.sum(axis=0)
+    nb = (nbs*wes).sum(axis=0)/wes.sum(axis=0)
```
Comment (ngbusca) [inline bin/do_xcf.py]: remove the weights, the number of pairs should be an intenger. Then nb = nbs.sum(axis=0)

File: py/pylya/xcf.py (hunk 4 above)
Comment (ngbusca) [inline py/pylya/xcf.py]: sorry, nb[:len(cnb)]+=cnb

**Comment by londumas** [comment]
If we want to keep the fact that it "nb" is integer, we have to specify it in the FITS file, here
    out.write([rp,rt,z,nb],names=['RP','RT','Z','NB'],header=head)

File: py/pylya/xcf.py [hunk 6]
```diff
@@ -43,6 +46,7 @@ def xcf(pix):
     rp = sp.zeros(np*nt)
     rt = sp.zeros(np*nt)
     z = sp.zeros(np*nt)
+    nb = sp.zeros(np*nt)
```
Comment (ngbusca) [inline py/pylya/xcf.py]: nb should be an int64: nb sp.zeros(np*nt,dtype=int64)

**Comment by londumas** [comment]
It is declared as an integer scipy array but we don't store it has is in the FITS file, so it is store as a float array:
nb = vac[1]['NB'][:]
print nb

-> [  141.   345.   621. ...,  6724.  6904.  7010.]

**Comment by ngbusca** [comment]
Can you test again whether it's saved as int or float?

**Comment by londumas** [comment]
I just did.
It is saved as a float I think because:

nb = vac[1]['NB'][:]
print nb

-> [ 141. 345. 621. ..., 6724. 6904. 7010.]

**Comment by londumas** [comment]
But sometime when dealing with big numbers, integers are tricky since they can loop around.
In my code I take only float.

**Comment by ngbusca** [comment]
When f is a float you can have f + 1 = f. Ints go negative, so you'll know that you went overboard. But int64 is much bigger than the standard int in c, so it should be ok.

**Comment by ngbusca** [comment]
I don't know how to specify the data type when writing with fitsio

**Comment by londumas** [comment]
me neither, but it would be good to do it.

**Comment by ngbusca** [comment]
I'll look into it

**Comment by londumas** [comment]
I got it, it is because of the
cfs=sp.array(cfs)
we thus have to do
nbs=cfs[:,5,:].astype(sp.int64)

**Comment by ngbusca** [comment]
indeed

**Pull request merged**